### PR TITLE
Feature/addvalidationfor generalinfosteep

### LINKED ISSUE #159: (SP: 2 ) Add validation for General info step
UserStory: #77, #78 
Blocked by: #154 

## Description
Add validation to required fields in General Info step for both student and tutor steppers

## Design 

Image

## Acceptance Criteria
- [x] First name and Last name fields must be required and show an error when user removed text from them.
- [x] Country and city have to be pulled from server on input focus
- [x] Length of additional info can't be more than 100 symbols and block texting when it reached limit
- [x] When there are errors in required fields stepper can't be finished

### PR DESCRIPTION
**Pull request by LeonidShamarin**
Add validation for General info step#159

- [x] First name and Last name fields must be required and show an error when user removed text from them.
- [x] Country and city have to be pulled from server
- [x] Length of additional info can't be more than 100 symbols and block texting when it reached limit
- [x] When there are errors in required fields stepper can't be finished

https://github.com/Space2Study-Project/Space2Study-FrontEnd/assets/122298176/d9e7e54f-9f53-414b-8c70-7b3589b7f126

### CODE REVIEW

**Review by marianamartsy — APPROVED** [review]

**Review by Mygal-Alex — APPROVED** [review]

**Review by ArtemHolikov — APPROVED** [review]
👍

**Review by AndriiNasadiuk — APPROVED** [review]

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 import useName from '~/hooks/use-name'
 import useCountryCityInfo from '~/hooks/use-country-city-info'
 
 import { styles } from '~/containers/tutor-home-page/general-info-step/GeneralInfoStep.styles'
 
-const GeneralInfoStep = ({ btnsBox }) => {
-  const { name, lastName } = useName()
+const GeneralInfoStep = ({ btnsBox, setIsFormValid }) => {
+  // const [firstName, setFirstName] = useState('')
+  // const [lastName, setLastName] = useState('')
```
Comment (vlad-khrychov) [inline src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx]: Delete comments before merge

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx [hunk 2]
```diff
@@ -70,15 +87,15 @@ const GeneralInfoStep = ({ btnsBox }) => {
         </Box>
         <Box data-testid='descField'>
           <TextField
-            inputProps={{ maxLength: 70 }}
+            inputProps={{ maxLength: 100 }}
```
Comment (vlad-khrychov) [inline src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx]: Here and on 98 row a same meaning of number 100 (before changing was 70). I suggest use some constant with this value (`const maxLength = 100`). And do not forgot use `useMemo` for it.

**Review by vlad-khrychov — COMMENTED** [review]
After changing you can merge this PR without additional approves.

**Pull request merged**